vmdb/app/models/miq_provision_virt_workflow.rb

```diff
@@ -605,9 +605,17 @@ def source_vm_rbac_filter(vms, condition = nil)
     template_msg += "  Passing inital template IDs: <#{vms.collect(&:id).inspect}>" unless vms.blank?
     $log.info "#{log_header} Checking for allowed templates for #{template_msg}"
     if filter_id.zero?
-      Rbac.search(search_options.merge(:targets => vms, :class => VmOrTemplate)).first
+      if vms.empty?
+        Rbac.search(search_options.merge(:class => VmOrTemplate, :results_format => :objects)).first
+      else
+        Rbac.filtered(vms, search_options.merge(:class => VmOrTemplate))
+      end
     else
-      MiqSearch.find(filter_id).search(vms, search_options).first
+      if vms.empty?
+        MiqSearch.find(filter_id).search([], search_options.merge(:results_format => :objects)).first
+      else
+        MiqSearch.find(filter_id).filtered(vms, search_options)
```

Comment:
I think this needs to call `Rbac.filtered` as there are no `MiqSearch.filtered` methods.